framework/src/media/audio/audio_manager.c

```diff
@@ -1223,6 +1360,32 @@ unsigned int get_input_bytes_to_frame(unsigned int bytes)
 	return pcm_bytes_to_frames(g_audio_in_cards[g_actual_audio_in_card_id].pcm, bytes);
 }
 
+unsigned int get_user_input_frames_to_byte(unsigned int frames)
+{
+	int buffer_size = 0;
+	audio_card_info_t *card;
+	if ((g_actual_audio_in_card_id < 0) || (frames == 0)) {
+		return 0;
+	}
+	card = &g_audio_in_cards[g_actual_audio_in_card_id];
+	buffer_size = frames * card->resample.user_channel * card->resample.format;
+	return buffer_size;
+}
+
+unsigned int get_user_input_bytes_to_frame(unsigned int bytes)
+{
+	int frame_size = 0;
+	audio_card_info_t *card;
+
+	if ((g_actual_audio_in_card_id < 0) || (bytes == 0)) {
+		return 0;
+	}
+	card = &g_audio_in_cards[g_actual_audio_in_card_id];
+	frame_size = bytes / card->resample.user_channel / card->resample.format;
```

Comment:
i feel zero check for the below variables should be performed before the above operation,
card->resample.user_channel & card->resample.format

 Else it may raise to divide by zero exception.

Reply:
The zero check is added.